qiita_db/test/test_software.py

```diff
@@ -494,8 +494,11 @@ def test_processing_jobs(self):
                     '6ad4d590-4fa3-44d3-9a8f-ddbb472b1b5f',
                     '063e553b-327c-4818-ab4a-adfe58e49860',
                     'ac653cb5-76a6-4a45-929e-eb9b2dee6b63']
-        exp = [qdb.processing_job.ProcessingJob(j) for j in exp_jids]
-        self.assertCountEqual(qdb.software.Command(1).processing_jobs, exp)
+
+        exp = set([qdb.processing_job.ProcessingJob(j) for j in exp_jids])
+        set_jobs = set(qdb.software.Command(1).processing_jobs)
```

Comment:
Sorting both the observed and expected output before comparing for equality would be a better choice. The disadvantage of taking sets() of both is that you implicitly remove any duplicate entries that might be present in qdb.software.Command(1).processing_jobs for whatever bug or other reason.

Reply:
We don't want there to be duplicates in qdb.software.Command(1).processing_jobs, right? 

Could I create a set for qdb.software.Command(1).processing_jobs and compare its length to the length of qdb.software.Command(1).processing_jobs, thus, checking for duplicates?

Reply:
Right, we don't want there to be but since this is test code I wouldn't assume that a recent change couldn't cause such an error to occur; it's just defensive programming.

IMHO comparing two ordered lists checks both the content and the lengths more elegantly, but I'll accept your alternative if you include a comment as to why you're checking the length.